pkg/app/AppCrudOperationService.go

```diff
@@ -289,6 +307,80 @@ func (impl AppCrudOperationServiceImpl) GetAppMetaInfo(appId int) (*bean.AppMeta
 	}
 	return info, nil
 }
+
+func (impl AppCrudOperationServiceImpl) GetHelmAppMetaInfo(appId string) (*bean.AppMetaInfoDto, error) {
+
+	// adding separate function for helm apps because for CLI helm apps, apps can be of form "1|clusterName|releaseName"
+	// In this case app details can be fetched using app name / release Name.
+
+	appIdSplitted := strings.Split(appId, "|")
+
+	app := &app.App{}
+	var err error
+
+	if len(appIdSplitted) > 1 {
+		appName := appIdSplitted[2]
+
+		app, err = impl.appRepository.FindAppAndProjectByAppName(appName)
+
+		if err != nil && err != pg.ErrNoRows {
+			impl.logger.Errorw("error in fetching app meta data", "err", err)
+			return nil, err
+		}
+
+		if app.Id == 0 {
+			app.AppName = appName
+		}
+
+	} else {
+
+		appIdInt, err := strconv.Atoi(appId)
+
+		if err != nil {
+			impl.logger.Errorw("error in converting appId to integer", "err", err)
+			return nil, err
+		}
+
+		InstalledApp, err := impl.installedAppRepository.GetInstalledApp(appIdInt)
+
+		app.Id = InstalledApp.AppId
+		app.AppName = InstalledApp.App.AppName
+		app.TeamId = InstalledApp.App.TeamId
+		app.Team.Name = InstalledApp.App.Team.Name
+		app.CreatedBy = InstalledApp.App.CreatedBy
+		app.Active = InstalledApp.App.Active
+
+		if err != nil {
```

Comment:
handle error before above operations. if no rows are found and InstalledApp is null then code will panic...always handle errors first